firebase/src/core/data/gateways/shell-gateway.ts

```diff
@@ -0,0 +1,20 @@
+import ShellError from "@faults/errors/shell-error";
+import { exec } from "child_process";
+
+/** Exposes access to perform shell operations. */
+export class ShellGateway {
+  /** Runs {@link command} in shell and returns an UTF-8 decoded string. */
+  async run(command: string): Promise<string> {
```

Comment:
Shouldn't this return the exit status code of the script?

Reply:
I don't think that returning the exit status make sense. We need to return the output because the gateway user may be interested in the shell output instead if it had success (exit 0) or not (exit 1). Also, the success of the command is measured by the Promise.resolve or Promise.reject, so I think we are good not using the exit status.